### ISSUE TITLE
O.Writable broken in 6.13.24

### ISSUE THREAD

**Issue by xaviergonz** [millsp/ts-toolbelt#134]
```ts
interface IM { n: number; }

function test<M extends IM>(m: M) {
  const fn1 = m.n; // OK
  const wm = m as O.Writable<M>;
  const fn2 = wm.n; // Property 'n' does not exist on type 'Writable<M, string | number | symbol, "flat">'.ts(2339)
}
```

used to work ok in 6.10.16 (and probably later).

**Comment by millsp** [comment]
Arf, sorry about this again. My tests pass but the metadata is somehow lost when working with generics. Now I'm wondering how this affects the library as a whole. Thanks for reporting. Working on it.

**Comment by xaviergonz** [comment]
No worries, always happy to help :)

**Comment by millsp** [comment]
Amazing, thank you. The fix will be out in a few moments.

**Comment by xaviergonz** [comment]
I'm afraid this is happening again in the latest version :(

**Comment by xaviergonz** [comment]
By the way, isn't there a way to test for this automatically by using // @ts-expect-error ?

**Comment by millsp** [comment]
You know what, I did not write tests for this, I wrote them for another utility `Patch` and `Merge` which I thought were the upstream problem... I'm so sorry, this is very unprofessional. I'm writing the same tests for all the other ones, I don't want to cause you problems anymore. Again, sorry!

**Comment by millsp** [comment]
Hey @xaviergonz. Thank you so much for this! You helped me uncover a very old, suspected bug that caused metadata loss because of (way) too much type distribution. I've added tests for all the concerned tools, I'm doing all I can to break this curse :laughing:.

I'm very sorry for the disturbance I caused. I hope you are still enjoying using ts-toolbelt and that you will stick to it in the future.

Many thanks!
